provider/pkg/helm/tool.go

```diff
@@ -0,0 +1,516 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// package helm contains code vendored from the upstream Helm project.
```

Comment:
Wondering what our maintenance strategy is for this vendored fork? Are we going to try and track upstream changes closely? If so, maybe a quick doc.go that describes how this code was vendored and the releavant changes made, if any?

Reply:
Also it appears that we have our own code within the vendored code in this file? If so, should we also include a Pulumi license to this file? Not sure what the correct procedure here is.

Reply:
I too was wondering about that but I'm sure we've done enough.

Reply:
+1 to headers being sufficient.

Part of the motivation behind my suggestion [here](https://github.com/pulumi/pulumi-kubernetes/pull/2947#discussion_r1588192130) was to help reduce non-upstream code, because otherwise things will start to get intermingled and harder to keep in sync.

It might make sense to break out our pulumi-specific stuff into its own package (pulumihelm or something) to make tracking upstream easier, but we don't need to do that now.